lua/neogen/init.lua

```diff
@@ -47,20 +37,10 @@ local autocmd = require("neogen.utilities.autocmd")
 ---@usage `require('neogen').setup({})` (replace `{}` with your `config` table)
 ---@toc_entry The setup function
 neogen.setup = function(opts)
-    -- Stores the user configuration globally so that we keep his configs when switching languages
-    neogen.user_configuration = opts or {}
-
-    neogen.configuration = vim.tbl_deep_extend("keep", neogen.user_configuration, neogen.configuration)
-
-    if neogen.configuration.enabled == true then
+    conf = config.setup(neogen.configuration, opts)
+    if conf.enabled then
         neogen.generate_command()
     end
-
-    -- Export module
-    _G.neogen = neogen
-
-    -- Force configuring current language again when doing `setup` call.
-    helpers.switch_language()
```

Comment:
I guess this is deleted because of the `__call` ?

Reply:
As a plugin, don't use `_G`, unless we have to.